x-pack/plugins/secops/public/containers/who_am_i/index.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { Query } from 'react-apollo';
+import { WhoAmIQuery } from '../../../common/graphql/types';
+
+import { whoAmIQuery } from './who_am_I.gql_query';
+
+interface WHoAmIArgs {
+  appName: string;
+}
+
+interface WHoAmIProps {
+  children: (args: WHoAmIArgs) => React.ReactNode;
+  sourceId: string;
+}
+
+export const WhoAmI = ({ children, sourceId }: WHoAmIProps) => (
+  <Query<WhoAmIQuery.Query, WhoAmIQuery.Variables>
+    query={whoAmIQuery}
+    fetchPolicy="no-cache"
+    notifyOnNetworkStatusChange
+    variables={{ sourceId }}
+  >
+    {({ data }) =>
+      children({
+        appName:
+          data && data.source && data.source.whoAmI ? data.source.whoAmI.appName : 'Who am I ?',
```

Comment:
I imagine that `graphql` does handle `null/undefined` with its schema's and we are failing fast with our queries for both requests and responses but this is a business use case example where `source` could be null, `whoAmI` could be null and `data.source.whoAmI.appName` could be null.

If so, I would recommend pulling in get from ` import const { get } from 'lodash/fp'` and then do this:
```
get(data, 'source.whoAmI.appName') ? data.source.whoAmI.appName : 'Who Am I?'
ref: https://lodash.com/docs/4.17.10#get
``` 
It looks a bit better.

For deep sets where something could be null/undefined along the way,  I prefer using the counter part of `set` as well:

https://lodash.com/docs/4.17.10#set
 
And typically use `/fp` to pull things in because it doesn't mutate anything.

Reply:
👍